src/main/java/feiner/fallingsand/Sand.java

```diff
@@ -34,15 +45,42 @@ public void put(int x, int y) {
 
     public void fall() {
         // moves all sand down one square
-        // if it reaches the floor or sand is below it
         for (int y = field.length - 2; y >= 0; y--) {
             for (int x = 0; x < field[y].length; x++) {
-                if (field[y][x] == 1 && field[y + 1][x] == 0) {
-                    // Next, move sand down
-                    field[y][x] = 0;
-                    field[y + 1][x] = 1;
+                if (field[y][x] == 1) {
+                    if (field[y + 1][x] == 0) {
+                        // does the sand fall straight down?
+                        field[y][x] = 0;
+                        field[y + 1][x] = 1;
+                        continue;
+                    }
+
+                    boolean rightFirst = random.nextBoolean();
+                    int direction1 = rightFirst ? 1 : -1;
+                    int direction2 = rightFirst ? -1 : 1;
+                    if (field[y + 1][x + direction1] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction1] = 1;
+                    } else if (field[y + 1][x + direction2] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction2] = 1;
+                    } else {
```

Comment:
What does this line do?

Reply:
Oh, I was not sure which line you were referring to. The else statement is there to handle any instances where the if and the else if is not met. I thought we should add that line to handle it to show that the space of (y,x) is already occupied.

Reply:
But we know that the space is already occupied.
Does line 68 do anything?

Reply:
No it doesn't really do anything, should I delete it?

Reply:
What do you think?

Reply:
Yes, I deleted it.